nucliadb_node/src/shards/shard_writer.rs

```diff
@@ -236,135 +240,196 @@ impl ShardWriter {
 
     #[measure(actor = "shard", metric = "open")]
     pub fn open(metadata: Arc<ShardMetadata>) -> NodeResult<ShardWriter> {
+        let span = tracing::Span::current();
         let shard_path = metadata.shard_path();
+
+        // fallback to default indexes while there are shards without the file
         let indexes = ShardIndexes::load(&shard_path).unwrap_or_else(|_| ShardIndexes::new(&shard_path));
 
-        // TODO: this call will generate the shard indexes file, as a lazy
-        // migration. When every shard has the file, this line should be
-        // removed
+        // This call will generate the shard indexes file, as a lazy migration.
+        // TODO: When every shard has the file, this line should be removed
         indexes.store()?;
 
+        let versions_path = metadata.shard_path().join(VERSION_FILE);
+        let versions = Versions::load(&versions_path)?;
+
         let tsc = TextConfig {
             path: indexes.texts_path(),
         };
+        let text_task = || Some(open_texts_writer(versions.texts, &tsc));
+        let info = info_span!(parent: &span, "Open texts index writer");
+        let text_task = || run_with_telemetry(info, text_task);
+
         let psc = ParagraphConfig {
             path: indexes.paragraphs_path(),
         };
+        let paragraph_task = || Some(open_paragraphs_writer(versions.paragraphs, &psc));
+        let info = info_span!(parent: &span, "Open paragraphs index writer");
+        let paragraph_task = || run_with_telemetry(info, paragraph_task);
+
+        let mut vector_tasks = vec![];
+        for (name, path) in indexes.iter_vectors_indexes() {
+            vector_tasks.push(|| {
+                // redefining here allows to get a copy of path without moving
+                // metadata
+                #[allow(clippy::redundant_locals)]
+                let path = path;
+                Some((name, open_vectors_writer(versions.vectors, &path, metadata.id())))
```

Comment:
An alternative (if you prefer it) is to extract the id from metadata and move that in:
```suggestion
            let id = metadata.id();
            vector_tasks.push(move || Some((name, open_vectors_writer(versions.vectors, &path, id))))
```